File: Chapter4/C#/question-4.5/binary-node.cs
using System.Text;

namespace C_ {
  class BinaryNode {
    public int value;
    public BinaryNode left;
    public BinaryNode right;
    public BinaryNode(int value) {
      this.value = value;
    }

    public string leftValue() {
      if (this.left != null) {
        return this.left.value.ToString();
      }
      return "";
    }

    public string rightValue() {
      if (this.right != null) {
        return this.right.value.ToString();
      }
      return "";
    }

    public override string ToString() {
      string str = "[BinaryNode value:";
      str = $"{str}{this.value.ToString()} l:{this.leftValue()} r:{this.rightValue()}]";
      return str;
    }
  }
}

File: Core/C++/DataStructures/Tree/binary-node.cpp
#include<string>
#include "binary-node.h"
using namespace std;

class BinaryNode {
  public:
    int value;

    BinaryNode* left;

    BinaryNode* right;

    BinaryNode(int value) {
      this->value = value;
    }

    string left_value() {
      if (left != NULL) {
        return to_string(left->value);
      }
      return "";
    }

    string right_value() {
      if (right != NULL) {
        return to_string(right->value);
      }
      return "";
    }

    string get_string() {
      return "[BinaryNode value:" + to_string(value) + ", l:" + left_value() + ", r:" + right_value() + "]";
    }

    ~BinaryNode() {
      delete this->left;
      delete this->right;
    }
};
